### PR TITLE
Optimize serialization and enable scala 2.13

### PR DESCRIPTION
**Pull request by Mefl**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -28,16 +28,6 @@ scalacOptions ++= Seq(
...
-  "-Ywarn-adapted-args",
-  "-Ywarn-dead-code",
-  "-Ywarn-inaccessible",
-  "-Ywarn-dead-code",
-  "-Ywarn-nullary-override",
-  "-Ywarn-nullary-unit",
-  "-Ywarn-numeric-widen",
-  "-Ywarn-unused",
-  "-Ywarn-unused-import",
-  "-Ywarn-value-discard",
```
Comment (KannarFr) [inline build.sbt:40]: Except the redundancy of some elemnts, why do you removed checks?

**Comment by KannarFr** [comment]
@Mefl Thanks for the help!

Can you please update the README example consequently?

**Comment by KannarFr** [comment]
(The import part).

File: build.sbt (hunk 1 above)
Comment (Mefl) [reply build.sbt:40]: they were deprecated in scala 2.12 and removed in scala 2.13

**Comment by Mefl** [comment]
good comment ;)
I worked on serialization (ie writer) part but I didn't changed to support the short format on input.

I propose to do the input part in a second PR tomorrow

File: build.sbt (hunk 1 above)
Comment (KannarFr) [reply build.sbt:40]: Oh ok.

**Comment by KannarFr** [comment]
Ok, so let's merge this one.

**Pull request merged**